documentcloud/documents/processing/ocr/main.py

```diff
@@ -71,34 +72,62 @@
 LARGE_IMAGE_SUFFIX = "-large"
 TXT_EXTENSION = ".txt"
 
+TESS_PDF_PREFIX = "ocr"
+
 
 def write_text_file(text_path, text, access):
     """Helper method to write a text file."""
     storage.simple_upload(text_path, text.encode("utf8"), access=access)
 
 
-def ocr_page(page_path, ocr_code="eng"):
+def ocr_page(page_path, upload_text_path, access, ocr_code="eng"):
     """Internal method to run OCR on a single page.
 
     Returns:
         The page text.
     """
-    # Capture the image as raw pixel data
+    # Initialize temporary files
+    tmp_files = {
+        "img": tempfile.mkstemp(suffix=".png")[1],
+        "pdf": tempfile.mkstemp()[1],
+        "text": tempfile.mkstemp()[1],
```

Comment:
I think this is fine, but curious why you use `mkstemp` instead of https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile